code-studio/src/js/components/progress/stage_progress.jsx

```diff
@@ -20,24 +109,26 @@ var StageProgress = React.createClass({
   },
 
   render() {
-    var progressDots = this.props.levels.map((level, index) => {
+    var progressDots = this.props.levels.map(level => {
+      var uid = level.uid || level.id.toString();
 
-      var innerClass = 'level_link ' + (level.status || 'not_tried');
-      if (level.kind === 'unplugged') {
-        innerClass += ' unplugged_level';
+      var dotStyle = Object.assign({}, style.dot.puzzle);
+      if (level.kind === 'assessment') {
+        Object.assign(dotStyle, style.dot.assessment);
       }
 
-      var outerClass = (level.kind === 'assessment') ? 'puzzle_outer_assessment' : 'puzzle_outer_level';
-      outerClass = (index === this.props.currentLevelIndex) ? 'puzzle_outer_current' : outerClass;
-
-      var isUnplugged = isNaN(level.title);
-      var dotStyle = {};
       if (this.props.largeDots) {
```

Comment:
maybe a comment somewhere explaining what large dots vs. small dots are?  must admit i'm not quite sure what the difference is.